### ISSUE TITLE
update KIND integration cluster version to use Kubernetes 1.23

### ISSUE THREAD

**Issue by listx** [kubernetes/test-infra#28710]
**What should be cleaned up or changed**:

Our CI jobs use KIND, and it currently uses version 1.20.2. But k8s-prow (https://prow.k8s.io/) uses version 1.23.15. Upgrade it to match 1.23, if possible (or at least move off of 1.20.2).

**Provide any links for context**:
https://github.com/kubernetes/test-infra/pull/28672#issuecomment-1424898618

**Comment by BenTheElder** [comment]
1.23 images should be available for current kind releases, listed in the KIND release notes

**Comment by BenTheElder** [comment]
1.23 is out of support now in Kubernetes, should be 1.24+

https://kubernetes.io/releases/

KIND still supports it for now, but I highly recommend sticking to a version upstream supports 1.24 is on the way out within O(months) as well.

**Comment by listx** [comment]
Looks like I worked on this two months ago and then forgot about it. We already merged #28811, which upgraded us from KIND 0.10.0 to 0.17.0. My intent at that time was to update the image reference manually, but because of the automated image bumps, we got upgraded to the new kubekins-e2e image which uses KIND 0.17.0.

And here's the thing --- because KIND 0.17.0 uses Kubernetes version 1.25.3 by default, that's the version we've been using in our integration tests for ~2 months now.

But now back to the original intent of this issue. We wanted to put us on Kubernetes version 1.23, mainly because that's what we run in production in prow.k8s.io. However, in light of your comment above I think it makes sense to run a minor version (or two) ahead of production because we want to be aware of any breaking changes that may come from newer versions on the horizon. Kubernetes itself should have pretty conservative deprecation policies WRT any API changes, and coupled with the fact that Prow doesn't itself use any advanced Kubernetes features, we are a "typical" Kubernetes application and we shouldn't need to constrain ourselves to *exactly* match the Kubernetes version we run in production.

For completeness, if we still wanted to run Kubernetes version 1.23, we would need to specify this version as an argument to KIND in our [test setup script](https://github.com/kubernetes/test-infra/blob/d0f1450930ccabb65237d89ed09bb3b0cc2cd752/prow/test/integration/setup-kind-cluster.sh#L119), by providing an `--image` flag as discussed [here](https://github.com/kubernetes-sigs/kind/blob/4855a7f135102d903e2e33569254a3db9d77c017/site/content/docs/user/quick-start.md?plain=1#L131).

Closing for now.

/close
